spec.bs

```diff
@@ -4862,6 +4922,13 @@ An <dfn>auction config</dfn> is a [=struct=] with the following items:
   :: A [=list=] of [=auction config=]s.
     Nested auctions whose results will also participate in a top level auction. Only the top level
     [=auction config=] can have component auctions.
+
+  : <dfn>deprecated render url replacements</dfn>
+  :: Null, a {{Promise}}, failure, or a [=list=] of [=ad keyword replacement=]s, each containing a 

```

Comment:
Could we get away with just setting to "null" in the failure case too? "failure" is typically reserved for URL parsing errors, and I'd like to see it not get too pervasively used outside of that, so I'm wondering if there is a meaningful distinction between null and failure here that we can collapse.

Reply:
We did this for all promise fields, specifically used failure for rejected promises to allow this [[=wait ... promises resolve=]](https://wicg.github.io/turtledove/#wait-until-configuration-input-promises-resolve) algorithm to return failure, which will finally let the runAdAuction throw a typeError. It's different from null, which is a valid value (when the field is missing, since the promise fields are optional). It does not have to be failure, but we need a unique state that's different from null to tell the promise is rejected, if we want to get rid of failure here.

[=handle-an-input-promise-in-configuration=] (https://wicg.github.io/turtledove/#handle-an-input-promise-in-configuration) (step 3).
[Here is an example] (https://github.com/WICG/turtledove/blob/main/spec.bs#L1005) of how we set auction signals to failure when the promise is rejected.